Adding field to existing measurement causes panic

In RC33, if you add a new field to an existing measurement, the server panics with:

```
panic: field does not exist for value2

goroutine 62 [running]:
github.com/influxdb/influxdb/tsdb.(*FieldCodec).EncodeFields(0xc20802bfd0, 0xc208125350, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/jason/go/src/github.com/influxdb/influxdb/tsdb/shard.go:525 +0x2fc
github.com/influxdb/influxdb/tsdb.(*Shard).validateSeriesAndFields(0xc2080f9640, 0xc20802be10, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/jason/go/src/github.com/influxdb/influxdb/tsdb/shard.go:372 +0xbc3
github.com/influxdb/influxdb/tsdb.(*Shard).WritePoints(0xc2080f9640, 0xc20802be10, 0x1, 0x1, 0x0, 0x0)
	/Users/jason/go/src/github.com/influxdb/influxdb/tsdb/shard.go:114 +0x75
github.com/influxdb/influxdb/tsdb.(*Store).WriteToShard(0xc20800b4c0, 0x11, 0xc20802be10, 0x1, 0x1, 0x0, 0x0)
	/Users/jason/go/src/github.com/influxdb/influxdb/tsdb/store.go:256 +0xf7
```

Reproduce with :
```
curl -G http://localhost:8086/query --data-urlencode "q=CREATE DATABASE foo"
curl -G http://localhost:8086/query --data-urlencode "q=CREATE RETENTION POLICY bar ON foo DURATION 1h REPLICATION 1 DEFAULT"
curl -X POST 'http://localhost:8086/write?db=foo&precision=s' -d 'measure1,tag1=1 value=1 1433760721'
curl -X POST 'http://localhost:8086/write?db=foo&precision=s' -d 'measure1,tag1=1 value=1,value2=1.0 1433760721'
```

will cause the panic.

See https://groups.google.com/forum/#!topic/influxdb/fOEd5a1RCIg

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,9 @@
+## v0.9.0-rc34 [unreleased]
+
+### Bugfixes
+
+- [#2869](https://github.com/influxdb/influxdb/issues/2869): Adding field to existing measurement causes panic
+
 ## v0.9.0-rc33 [2015-06-09]
 
 ### Bugfixes

--- a/tsdb/shard.go
+++ b/tsdb/shard.go
@@ -369,11 +369,6 @@ func (s *Shard) validateSeriesAndFields(points []Point) ([]*seriesCreate, []*fie
 					return nil, nil, fmt.Errorf("input field \"%s\" is type %T, already exists as type %s", name, value, f.Type)
 				}
 
-				data, err := mf.codec.EncodeFields(p.Fields())
-				if err != nil {
-					return nil, nil, err
-				}
-				p.SetData(data)
 				continue // Field is present, and it's of the same type. Nothing more to do.
 			}
 

--- a/tsdb/shard_test.go
+++ b/tsdb/shard_test.go
@@ -75,6 +75,62 @@ func TestShardWriteAndIndex(t *testing.T) {
 	}
 }
 
+func TestShardWriteAddNewField(t *testing.T) {
+	tmpDir, _ := ioutil.TempDir("", "shard_test")
+	defer os.RemoveAll(tmpDir)
+	tmpShard := path.Join(tmpDir, "shard")
+
+	index := NewDatabaseIndex()
+	sh := NewShard(index, tmpShard)
+	if err := sh.Open(); err != nil {
+		t.Fatalf("error openeing shard: %s", err.Error())
+	}
+	defer sh.Close()
+
+	pt := NewPoint(
+		"cpu",
+		map[string]string{"host": "server"},
+		map[string]interface{}{"value": 1.0},
+		time.Unix(1, 2),
+	)
+
+	err := sh.WritePoints([]Point{pt})
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+
+	pt = NewPoint(
+		"cpu",
+		map[string]string{"host": "server"},
+		map[string]interface{}{"value": 1.0, "value2": 2.0},
+		time.Unix(1, 2),
+	)
+
+	err = sh.WritePoints([]Point{pt})
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+
+	if !reflect.DeepEqual(index.names, []string{"cpu"}) {
+		t.Fatalf("measurement names in shard didn't match")
+	}
+	if len(index.series) != 1 {
+		t.Fatalf("series wasn't in index")
+	}
+	seriesTags := index.series[string(pt.Key())].Tags
+	if len(seriesTags) != len(pt.Tags()) || pt.Tags()["host"] != seriesTags["host"] {
+		t.Fatalf("tags weren't properly saved to series index: %v, %v", pt.Tags(), index.series[string(pt.Key())].Tags)
+	}
+	if !reflect.DeepEqual(index.measurements["cpu"].TagKeys(), []string{"host"}) {
+		t.Fatalf("tag key wasn't saved to measurement index")
+	}
+
+	if len(index.measurements["cpu"].FieldNames()) != 2 {
+		t.Fatalf("field names wasn't saved to measurement index")
+	}
+
+}
+
 func BenchmarkWritePoints_NewSeries_1K(b *testing.B)   { benchmarkWritePoints(b, 38, 3, 3, 1) }
 func BenchmarkWritePoints_NewSeries_100K(b *testing.B) { benchmarkWritePoints(b, 32, 5, 5, 1) }
 func BenchmarkWritePoints_NewSeries_250K(b *testing.B) { benchmarkWritePoints(b, 80, 5, 5, 1) }

